tests/codeception/functional/MegaMenuCest.php

```diff
@@ -0,0 +1,108 @@
+<?php
+
+use Drupal\Core\Url;
+
+/**
+ * Class MegaMenuCest.
+ *
+ * @group install
+ */
+class MegaMenuCest {
+
+  /**
+   * Every main menu item should not error.
+   */
+  public function testMegaMenu(FunctionalTester $I) {
+
+    $I->logInWithRole('administrator');
+    $top_level = $I->createEntity([
+      'title' => 'Top Level Page',
+      'type' => 'hs_basic_page',
+    ]);
+
+    $I->amOnPage($top_level->toUrl('edit-form')->toString());
+    $I->click('.menu-link-form summary');
+    $I->checkOption('Provide a menu link');
+    $I->fillField('Menu link title', 'Top Level Page');
+    $I->scrollTo(['css' => '.form-submit']);
+    $I->click('Save');
+
+    $second_level = $I->createEntity([
+      'title' => 'Second Level Page',
+      'type' => 'hs_basic_page',
+    ]);
+    $I->amOnPage($second_level->toUrl('edit-form')->toString());
+    $I->click('.menu-link-form summary');
+    $I->checkOption('Provide a menu link');
+    $I->fillField('Menu link title', 'Second Level Page');
+    $I->scrollTo(['css' => '.form-item-menu-menu-parent'], 0, -100);
+    $I->selectOption('Parent link', '-- Top Level Page');
+    $I->wait(2);
```

Comment:
A better thing is to `$I->waitForText()`, `$I->waitForElementVisible()`, or `$I->waitForAjaxToFinish()`. It will speed up the test and in case something takes longer than 2 seconds, it won't fail. Please replace as many `wait` methods with something better if possible.

Reply:
Removed standard waits, if needing to be replaced, I replaced with ```waitForText```